R/setup.R

```diff
@@ -0,0 +1,66 @@
+# Set various global options for each pipeline stage
+options(
+  tidymodels.dark = TRUE, # Tidymodels CV output compatible with dark theme
+  dplyr.summarise.inform = FALSE, # Disable messages when aggregating
+  scipen = 99, # Disable scientific notation
+  knitr.kable.NA = "", # Replace NA with empty string in knitr::kable
+  java.parameters = "-Xmx10g" # Pre-allocate memory for Java JDBC driver
+)
+
+# Load R pipeline dependencies from the Depends: key in the DESCRIPTION file.
+# Any new pipeline dependencies should be added there
+suppressPackageStartupMessages({
+  purrr::walk(
+    strsplit(yaml::read_yaml("DESCRIPTION")$Depends, ", ")[[1]],
+    library,
+    character.only = TRUE
+  )
+})
+
+# Resolve package namespace conflicts, preferring the library::function pair
+# shown over other functions with the same name from different libraries
+conflicts_prefer(
+  dplyr::filter,
+  dplyr::lag,
+  dplyr::pull,
+  dplyr::slice,
+  lubridate::duration,
+  purrr::flatten,
+  purrr::set_names,
+  purrr::splice,
+  purrr::when,
+  recipes::step,
+  rlang::is_empty,
+  tictoc::push,
+  utils::timestamp,
+  .quiet = TRUE
+)
+
+# Load helpers for model dictionary, data loading, and other misc functions
+source(here::here("R", "helpers.R"))
```

Comment:
**[Question, non-blocking]** Given that we seem to walk the `R` directory to load this script in each pipeline stage, do we actually need to source `R/helpers.R` here manually, or can we count on the `walk()` call doing so?

https://github.com/ccao-data/model-res-avm/blob/fe5a790208f73fd273536ebe930b6609f7f1344e/pipeline/00-ingest.R#L12-L13

Reply:
We need to in order to load `model_file_dict()` within the setup script, else loading it is dependent on the walk order.